.github/workflows/docker.yml

```diff
@@ -0,0 +1,87 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    branches:
+      - "master"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: PHP ${{ matrix.php-version }}, Fixer ${{ github.ref == 'refs/heads/master' && 'edge' || github.ref_name }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            php-suffix: '-php7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            php-suffix: '-php8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            php-suffix: '-php8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            php-suffix: '-php8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            php-suffix: '-php8.3'
+            alpine-version: '3.18'
+          # This is the latest supported PHP Version and will be provided without -phpX.X suffix
+          - php-version: '8.3'
+            php-suffix: ''
+            alpine-version: '3.18'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Make image lowercase
+        run: |
+          echo IMAGE_NAME_LOWER=$(echo ${{ env.IMAGE_NAME }} | tr '[:upper:]' '[:lower:]') >> ${GITHUB_ENV}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          flavor: |
+            latest=false
+            suffix=${{ matrix.php-suffix }}
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LOWER }}
+          # "edge" will only build from master, "semver" only from tags
+          tags: |
+            type=edge,branch=master
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            type=semver,pattern={{major}}
+            type=semver,pattern=latest
```

Comment:
```suggestion
            type=edge,branch=master,suffix=-php${{ matrix.php-version }}
            type=semver,pattern={{version}},suffix=-php${{ matrix.php-version }}
            type=semver,pattern={{major}}.{{minor}},suffix=-php${{ matrix.php-version }}
            type=semver,pattern={{major}},suffix=-php${{ matrix.php-version }}
            type=semver,pattern=latest,suffix=-php${{ matrix.php-version }}
            type=raw,value=latest,enable=${{ '8.3' == matrix.php-version && 'tag' == github.ref_type }}
```

Reply:
I think this will not lead to the expected result. There would be no more tags `3`, `3.14` `3.14.0`. For that we would need to duplicate the other semver lines, too with an enable condition.

Reply:
No, it works as expected - tested [here](https://github.com/Wirone/PHP-CS-Fixer/actions/runs/7277270285/job/19828907706) 🙂. `latest` tag is added only in 8.3 ([here](https://github.com/Wirone/PHP-CS-Fixer/actions/runs/7277270285/job/19828909493#step:5:55)), other PHP versions have full set of semver tags and `latest-phpX.Y`. I did not test `edge`, though.

Reply:
So you don't want to have `3`, `3.14` and `3.14.0` tags? I can't see them at your example. Then the documentation needs to be adapated, too.

Reply:
Ah, semver without PHP version 🤔? I am not sure. I did not consider them in [my initial reply](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7555#pullrequestreview-1781457630) and I think these versions would be confusing because of lack of PHP version context.

Reply:
That's what I meant with "comfort", just use a version and up you go. If we have a `latest` version without explicit php version, then it would be consequent to also have this for semver tags. Otherwise I would suggest to skip the `latest` tag and only have `latest-phpX` tags, that's also fine for me.

Reply:
You're right 👍. Then let's remove `latest` for now. It's easier to introduce short semver  and `latest` later rather than dropping it after people started to use it.